pallets/pools/src/lib.rs

```diff
@@ -51,22 +52,48 @@ pub trait TrancheToken<T: Config> {
 pub struct TrancheInput<Rate> {
 	pub interest_per_sec: Option<Rate>,
 	pub min_risk_buffer: Option<Perquintill>,
-	pub seniority: Option<u32>,
+	pub seniority: Option<Seniority>,
 }
 
 #[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, Default, TypeInfo)]
 pub struct Tranche<Balance, Rate> {
 	pub interest_per_sec: Rate,
 	pub min_risk_buffer: Perquintill,
-	pub seniority: u32,
+	pub seniority: Seniority,
 
 	pub outstanding_invest_orders: Balance,
 	pub outstanding_redeem_orders: Balance,
 
 	pub debt: Balance,
 	pub reserve: Balance,
 	pub ratio: Perquintill,
-	pub last_updated_interest: u64,
+	pub last_updated_interest: Moment,
+}
+
+/// A type alias for the Tranche weight calculation
+type NumTranches = u32;
+
+impl<Balance, Rate> TrancheWeigher for Tranche<Balance, Rate>
+where
+	Balance: From<u128>,
+{
+	type Weight = (Balance, Balance);
```

Comment:
Should this just be `(u128, u128)`, since we're not holding any sort of Balance here?

Reply:
Yeah, I don't really like this neither. But i am also not a friend of raw types. But I would be fine with this optionX